File: backend/Payments/src/Payments.Application/InvoiceApi/InvoiceDto.cs
namespace Payments.Application.Common.Models;

public class InvoiceDto
{
    public long Id { get; set; }
    public string Number { get; set; }

    public long BuyerId { get; set; }
    public long SellerId { get; set; }
    public decimal Total { get; set; }
}

File: backend/InvoicingPython/app/api/invoices/models/dto/invoice_dto.py
from datetime import datetime
from decimal import Decimal
from typing import Optional

import attr

from .company_dto import CompanyDTO
from .invoice_item_dto import InvoiceItemDTO


@attr.s(auto_attribs=True)
class InvoiceDTO:
    id: int
    number: int
    createdDate: datetime
    dueDate: datetime
    notes: str
    currency: str
    buyerId: int
    sellerId: int
    total: Decimal
    approvalStatus: int
    type: str
    paymentStatus: str
    category: str
    templatePreview: str
    company: Optional[CompanyDTO]
    items: list[InvoiceItemDTO]


@attr.s(auto_attribs=True)
class InvoiceNumberDTO:
    newInvoiceNumber: int
